docs/architecture/adr-044-lite-client-with-weak-subjectivity.md

```diff
@@ -53,22 +54,25 @@ network or when a light client that has been offline for longer than the
 unbonding period connects to the network. Specifically, the node needs to
 initialize the following structure before syncing from user input:
 
-```
+```go
 type TrustOptions struct {
-    // Required: only trust commits up to this old.
-    // Should be equal to the unbonding period minus some delta for evidence reporting.
-    TrustPeriod time.Duration `json:"trust-period"`
-
-    // Option 1: TrustHeight and TrustHash can both be provided
-    // to force the trusting of a particular height and hash.
-    // If the latest trusted height/hash is more recent, then this option is
-    // ignored.
-    TrustHeight int64  `json:"trust-height"`
-    TrustHash   []byte `json:"trust-hash"`
-
-    // Option 2: Callback can be set to implement a confirmation
-    // step if the trust store is uninitialized, or expired.
-    Callback func(height int64, hash []byte) error
+	// Required: only trust commits up to this old.
+	// Should be equal to the unbonding period minus some delta for evidence reporting.
+	TrustPeriod time.Duration `json:"trust-period"`
+
+	// Required: validator whom we've got the TrustHeight/Hash from
+	ValidatorAddress types.Address `json:"validator-address"`
```

Comment:
Had to add validator address for evidence handling & submission

Reply:
This means full nodes can't publish hashes?

Reply:
Yes. We can't slash full nodes because they aren't bonded.